lib/stat_tracker.rb

```diff
@@ -21,6 +21,39 @@ def initialize(files)
     end
   end
 
+  def highest_scoring_home_team
+    # @game_teams.csv
+    # 1. find total amount of goals scored by each home team and keep track of the amount of games
+    # hash {[team_id #] => [total goals, total games played by home team]}
+    # 2. get hash.values.max
+    # 3. ^find the correlary teamid
+    # iterate thru @teams.csv & match hash.values.max to team_id # 
+      # then output teamName
+
+    # 1.
+    total_goals = @game_teams.each_with_object({}) do |game, hash|
+      # get the curr val for game.team_id or if doesn't exist: set it to [0,0]
+      hash[game.team_id] = hash[game.team_id] || [0, 0]
+      if game.hoa == "home"
+        hash[game.team_id] = [game.goals + hash[game.team_id][0], hash[game.team_id][1] + 1]
+      end
+    end
+
+    # 2.
+    avg_goals = total_goals.transform_values do |value|
+      (value[0] / value[1].to_f).round(4)
+    end
+    highest_avg_goals = avg_goals.values.max
+
+    # 3. 
+    highest_team_id = avg_goals.key(highest_avg_goals)
+
+    # 4.
+    @teams.each do |team| 
+      return team.team_name if team.team_id == highest_team_id
```

Comment:
funny how we came to the same exact conclusion. only difference is I used find rather than which which does the same thing in this instance.